### PR TITLE
Added pytest mark to skip based on server version

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Added pytest mark to skip based on server version

(testing is very basic, as this is a bit hard to test in this context without adding a 'skipped' testcase)

Part of https://github.com/inmanta/lsm/issues/320

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [X] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

### CODE REVIEW

File: changelogs/unreleased/pytest-mark-to-skip-based-on-version.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+description: Added pytest mark to skip based on server version
+change-type: minor
```
Comment (arnaudsjs) [inline changelogs/unreleased/pytest-mark-to-skip-based-on-version.yml:3]: Why is this a minor change?

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: tests/utils.py [hunk 2]
```diff
@@ -297,3 +304,23 @@ def get_resource(version: int, key: str = "key1", agent: str = "agent1", value:
...
+
+def product_version_lower_then(version: str) -> bool:
+    return parse_version(get_product_meta_data().version) < parse_version(version)
+
+
+def mark_only_for_version_higher_then(version: str) -> "MarkDecorator":
+    current = get_product_meta_data().version
+    return pytest.mark.skipif(
+        product_version_lower_then(version),
+        reason=f"This test is only intended for version larger then {version} currently at {current}",
```
Comment (sanderr) [inline tests/utils.py:325]:
```suggestion
def product_version_lower_than(version: str) -> bool:
    return parse_version(get_product_meta_data().version) < parse_version(version)


def mark_only_for_version_higher_than(version: str) -> "MarkDecorator":
    current = get_product_meta_data().version
    return pytest.mark.skipif(
        product_version_lower_than(version),
        reason=f"This test is only intended for version larger than {version} currently at {current}",
```

File: changelogs/unreleased/pytest-mark-to-skip-based-on-version.yml (hunk 1 above)
Comment (wouterdb) [reply changelogs/unreleased/pytest-mark-to-skip-based-on-version.yml:3]: It adds something to the api, but on the other hand, only to the testing api,...

Comment (wouterdb) [reply changelogs/unreleased/pytest-mark-to-skip-based-on-version.yml:3]: if you want, I can make it a patch.

**Review by arnaudsjs — APPROVED** [review]

File: tests/utils.py [hunk 3]
```diff
@@ -297,3 +304,23 @@ def get_resource(version: int, key: str = "key1", agent: str = "agent1", value:
...
+    bootloader = InmantaBootloader()
+    context = bootloader.load_slices()
+    return context.get_product_metadata()
+
+
+def product_version_lower_than(version: str) -> bool:
+    return parse_version(get_product_meta_data().version) < parse_version(version)
+
+
+def mark_only_for_version_higher_than(version: str) -> "MarkDecorator":
```
Comment (arnaudsjs) [inline tests/utils.py:321]: This name indicates that the test case will run for all versions strictly larger than the given version, but looking at the implementation, it seems that the test case will also run when the versions match. I think the implementation is incorrect. Shouldn't we skip the test case when the given version is smaller then or equal to the given version?

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: tests/utils.py (hunk 3 above)
Comment (wouterdb) [reply tests/utils.py:321]: oh my, good catch, thx

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**